### ISSUE TITLE
Dump error => pq: internal error: Non-nullable column "push_user:token" with no value! Index scanned was "primary" with the index key columns (uid) and the values ('982e4ffc-e4a0-4a42-939b-cce190961686') Failed running "dump"

### ISSUE THREAD

**Issue by eymerich** [cockroachdb/cockroach#41957]
I'm trying to execute a backup of my database by commandline:
`cockroach dump databasename --certs-dir=certs --host=127.0.0.1 > dump`

Program exits with the following error:

> Error: pq: internal error: Non-nullable column "push_user:token" with no value! Index scanned was "primary" with the index key columns (uid) and the values ('982e4ffc-e4a0-4a42-939b-cce190961686')
> Failed running "dump"

CRDB logs reports:

> E191028 16:23:45.875849 66523791 sql/sqltelemetry/report.go:39  [intExec=create-stats] encountered internal error:
> sql/row/fetcher.go:1336 in finalizeRow(): (XX000) internal error: Non-nullable column "push_user:token" with no value! Index scanned was "primary" with the index key columns (uid) and the values ('982e4ffc-e4a0-4a42-939b-cce190961686')
> -- detail --
> fetcher.go:1336: Non-nullable column "%s:%s" with no value! Index scanned was %q with the index key columns (%s) and the values (%s) | string; string; string; string; string
> github.com/cockroachdb/cockroach/pkg/sql/row/fetcher.go:1336: in finalizeRow()
> github.com/cockroachdb/cockroach/pkg/sql/row/fetcher.go:1078: in NextRow()
> github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/tablereader.go:165: in Next()
> github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/tablereader.go:294: in Next()
> github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sampler.go:192: in mainLoop()
> github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/sampler.go:170: in Run()
> github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:566: in func1()
> 
> -- detail --
> statement: CREATE STATISTICS _ FROM [73] WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME _
> E191028 16:23:46.025581 66523791 util/log/crash_reporting.go:546  [intExec=create-stats] Reported as error 15a429f614614d3d99aeee1c1ec544d3
> W191028 16:23:46.025655 66516435 sql/stats/automatic_stats.go:450  failed to create statistics on table 73: create-stats: internal error: Non-nullable column "push_user:token" with no value! Index scanned was "primary" with the index key columns (uid) and the values ('982e4ffc-e4a0-4a42-939b-cce190961686')

I have tested with schema and data options, schema went well, data same error as before.

This is the table schema:
```
CREATE TABLE push_user (
	uid UUID NOT NULL DEFAULT gen_random_uuid(),
	token STRING NOT NULL,
	country STRING NOT NULL,
	domain INT8 NULL,
	dt_created TIMESTAMP NOT NULL,
	dt_last_proc TIMESTAMP NULL,
	sent_interval INT8 NOT NULL DEFAULT 1:::INT8,
	ip INET NULL,
	useragent STRING NULL,
	source STRING NULL,
	os STRING NULL,
	browser STRING NULL,
	CONSTRAINT "primary" PRIMARY KEY (uid ASC),
	UNIQUE INDEX push_users_token_key (token ASC),
	INDEX push_user_dt_last_proc_country_domain_idx (dt_last_proc ASC, country ASC, domain ASC),
	FAMILY f1 (uid, token, country, domain, dt_created, sent_interval, ip, useragent, source, os, browser),
	FAMILY f2 (dt_last_proc),
	CONSTRAINT check_domain CHECK (domain > 0)
);
```

Record with the uid stated in the error has all cells values not NULL and is actually querable:
`cockroach sql --certs-dir=certs --host=127.0.0.1 --database=databasename --execute="select * from push_user where uid = '982e4ffc-e4a0-4a42-939b-cce190961686'" > row.csv`
Result is attached.
[row.txt](https://github.com/cockroachdb/cockroach/files/3779851/row.txt)

We have recently started the adoption of CRDB within our infrastructure, whole database is 1Gb and two tables, I have taken 3 dumps successfully when the table had different keys, in place of the composite index push_user_dt_last_proc_country_domain_idx there was only dt_last_proc, this is the first dump that I tried to take after the schema change, coindicence?

**Comment by jordanlewis** [comment]
@eymerich, sorry we took so long to get back to you on this. Which schema change did you try before you saw the error?

**Comment by eymerich** [comment]
Dropped  INDEX push_user_dt_last_proc_idx (dt_last_proc ASC) an added INDEX push_user_dt_last_proc_country_domain_idx (dt_last_proc ASC, country ASC, domain ASC).

The error appear only for query  SELECT without a LIMIT, if I use LIMIT (whatsoever: <, = or > than the current number of records) error does not show up.

I have generated an export with that query and token field is never without data.

**Comment by ricardocrdb** [comment]
Hey @eymerich 

Apologies for the delay in response. What version were you running when you ran into this issue? Have you attempted this same dump with the latest version 19.2?

**Comment by yuzefovich** [comment]
Closed by #54543
